### PR TITLE
Update virtual port docs

### PR DESCRIPTION
**Pull request by jonbarrow**
Adds some more details about virtual ports, including renaming the "port number" to "stream ID" as seen in the real lib, moving around the stream type table, and mentioning how the streams control aspects of the connections

### CODE REVIEW

File: docs/prudp.md [hunk 1]
```diff
@@ -259,26 +259,29 @@ The CONNECT acknowledgement packet contains a [Buffer] with the following data:
...
-|-------------|------------------|-------------|----------|
-| 1           | DO               | 7           | NATEcho  |
-| 2           | RV               | 8           | Routing  |
-| 3           | OldRVSec         | 9           | Game     |
-| 4           | SBMGMT           | 10          | RVSecure |
-| 5           | NAT              | 11          | Relay    |
-| 6           | SessionDiscovery |             |          |
+| Stream type | Name             | Details                                                              |
+|-------------|------------------|----------------------------------------------------------------------|
+| 1           | DO               | Unknown purpose                                                      |
```
Comment (DaniElectra) [inline docs/prudp.md]: DO is used for P2P on NetZ

File: docs/prudp.md [hunk 2]
```diff
@@ -259,26 +259,29 @@ The CONNECT acknowledgement packet contains a [Buffer] with the following data:
...
+| 2           | RV               | Unknown purpose                                                      |
+| 3           | OldRVSec         | Original "secure" stream type. Used by NEX 1 and QRV clients         |
+| 4           | SBMGMT           | Unknown purpose                                                      |
+| 5           | NAT              | Unknown purpose                                                      |
```
Comment (DaniElectra) [inline docs/prudp.md]: NAT is used for NAT traversal between users on NetZ

File: docs/prudp.md (hunk 1 above)
Comment (jonbarrow) [reply docs/prudp.md]: This is the "Duplicate Object" thing correct?

Comment (DaniElectra) [reply docs/prudp.md]: Yes!

File: docs/prudp.md [hunk 3]
```diff
@@ -259,26 +259,29 @@ The CONNECT acknowledgement packet contains a [Buffer] with the following data:
 | Uint32 | Response check value + 1 |
 
 ### Virtual ports
-When multiple PRUDP connections are made to the same address, NEX doesn't create a new socket for each connection. Instead, it uses a single socket to create multiple PRUDP connections. To distinguish between connections, each packet contains a source and destination port.
+When multiple PRUDP connections are made to the same address, NEX doesn't create a new socket for each connection. Instead, it uses a single socket to create multiple PRUDP connections. To distinguish between connections, each packet contains a source and destination "virtual port". A "virtual port" is made of 2 parts, a stream type and stream ID. The client and server stream types will always match, though their IDs may not. How these are encoded depends on the protocol version. Each stream may be configured differently. Aspects of the connection such as retransmission rates, the encryption and compression algorithms, etc, are all controlled by these streams.
```
Comment (DaniElectra) [inline docs/prudp.md]:
```suggestion
When multiple PRUDP connections are made to the same address, NEX doesn't create a new socket for each connection. Instead, it uses a single socket to create multiple PRUDP connections. To distinguish between connections, each packet contains a source and destination "virtual port". A "virtual port" is made of 2 parts, a stream type and stream ID. The client and server stream types will always match, though their IDs may not. Each stream may be configured differently: aspects of the connection such as retransmission rates, the encryption and compression algorithms, etc, are all controlled by these streams. How these are encoded depends on the protocol version:
```

**Review by DaniElectra — APPROVED** [review]
Looks good

**Pull request merged**